pkg/kubectl/apply/parse/openapi.go

```diff
@@ -104,6 +104,12 @@ func (v *baseSchemaVisitor) VisitKind(*proto.Kind) {
 	v.Err = fmt.Errorf("Kind type not expected")
 }
 
+// VisitArbitrary implements openapi
+func (v *baseSchemaVisitor) VisitArbitrary(schema *proto.Arbitrary) {
+	v.Kind = "arbitrary"
+	v.Err = fmt.Errorf("Arbitrary type not expected")
```

Comment:
@pwittrock is that OK?

Reply:
This code is a re-implementation of the existing "3-way merge" logic that we have in kubectl. It's unfortunately not used yet (probably will never be). Since it can't easily be triggered, I'm tempted to assume you haven't tested this specifically with your new type.

In order to make this work, you would have to change other parts of the code. For example, the "baseSchemaVisitor" just creates "failure" for each methods by default. Each subtype then reimplements one of these methods for the valid "merge" use-case (primitive merges with primitives, array merges with array, etc). Unfortunately, you haven't created a subtype for "Arbitrary" (that subtype would know how to merge Arbitrary types with Arbitrary types). In other words, this code is missing the logic to merge arbitrary types.

It's not terrible since this code is not really used anyway, but the real question is:
How are we supposed to merge "Arbitrary types"?